### PR TITLE
Constraints file

### PR DESCRIPTION
**Pull request by AlexanderDokuchaev**
### Changes

- Use constraints file to set versions of packages in one place
- Depreciated message about installation extra requirements by `pip install nncf[torch]`
- Fix System requirements of python version (actual python>=3.8)

### Related tickets

134503

### CODE REVIEW

**Comment by KodiaqQ** [comment]
@AlexanderDokuchaev, do you have successful runs for the **test_examples**, **conformance run** (for a few models), and **onnx e2e**? Don't we need to run e2e's for PyTorch and Tensorflow as well?

**Comment by alexsu52** [comment]
@AlexanderDokuchaev I would recommend to present this approach on technical sync. 
cc '@MaximProshin.

**Comment by AlexanderDokuchaev** [comment]
manual/job/post_training_quantization/334
nightly/job/e2e_onnx_model_zoo/631
nightly/job/test_examples/313/

**Review by alexsu52 — APPROVED** [review]
LGTM

**Comment by MaximProshin** [comment]
@AlexanderDokuchaev , please also run SDL checks on this PR.

File: tests/post_training/requirements.txt [hunk 1]
```diff
@@ -1,21 +1,19 @@
...
-pandas==2.0.3
+optimum-intel==1.15.2
+optimum==1.17.1
+scikit-learn==1.2.2
+soundfile==0.12.1
 tensorboard==2.13.0
 tensorflow-io==0.32.0
 timm==0.9.2
-scikit-learn==1.2.2
+transformers==4.38.2
```
Comment (KodiaqQ) [inline tests/post_training/requirements.txt:18]: Can we also add the **transformers** package version to the constraints file?

Comment (AlexanderDokuchaev) [reply tests/post_training/requirements.txt:18]: We decided to add only backends and pytest in constrains file now.
After constraints file will be expanded if it is useful.

**Comment by AlexanderDokuchaev** [comment]
@MaximProshin sdl/job/nightly_trigger/129 - protex fails by not related issue

**Comment by MaximProshin** [comment]
> sdl/job/nightly_trigger/129

we'll need to check Protex as well when it's recovered.

**Comment by AlexanderDokuchaev** [comment]
@MaximProshin  sdl/job/Protex_Scan_ABI/707/ -passed

**Review by MaximProshin — APPROVED** [review]

**Pull request merged**